Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/interpolated_corrected_curvatures.h

```diff
@@ -0,0 +1,1214 @@
+// Copyright (c) 2022 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Hossam Saeed
+//
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_INTERPOLATED_CORRECTED_CURVATURES_H
+#define CGAL_POLYGON_MESH_PROCESSING_INTERPOLATED_CORRECTED_CURVATURES_H
+
+#include <CGAL/license/Polygon_mesh_processing/interpolated_corrected_curvatures.h>
+
+#include <CGAL/assertions.h>
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#include <CGAL/Polygon_mesh_processing/measure.h>
+#include <CGAL/Named_function_parameters.h>
+#include <CGAL/property_map.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <Eigen/Eigenvalues>
+
+#include <numeric>
+#include <queue>
+#include <unordered_set>
+
+namespace CGAL {
+
+namespace Polygon_mesh_processing {
+
+/**
+  * \ingroup PMP_corrected_curvatures_grp
+  *
+  * \brief a struct for storing principal curvatures and directions.
+  *
+  * @tparam GT is the geometric traits class, model of `Kernel`.
+*/
+template<typename GT>
+struct Principal_curvatures_and_directions {
+
+  /// min curvature magnitude
+  typename GT::FT min_curvature;
+
+  /// max curvature magnitude
+  typename GT::FT max_curvature;
+
+  /// min curvature direction vector
+  typename GT::Vector_3 min_direction;
+
+  /// max curvature direction vector
+  typename GT::Vector_3 max_direction;
+
+  Principal_curvatures_and_directions() {
+    min_curvature = 0;
+    max_curvature = 0;
+    min_direction = typename GT::Vector_3(0, 0, 0);
+    max_direction = typename GT::Vector_3(0, 0, 0);
+  }
+
+  Principal_curvatures_and_directions(
+    typename GT::FT min_curvature,
+    typename GT::FT max_curvature,
+    typename GT::Vector_3 min_direction,
+    typename GT::Vector_3 max_direction
+  ) {
+    this->min_curvature = min_curvature;
+    this->max_curvature = max_curvature;
+    this->min_direction = min_direction;
+    this->max_direction = max_direction;
+  }
+};
+
+namespace internal {
+
+template<typename PolygonMesh, typename GT>
+typename GT::FT average_edge_length(const PolygonMesh& pmesh) {
```

Comment:
@sloriot should that not become a function in PMP::measure ?   And in a way that it works also with EPECK?

Reply:
Moved to #7859 7859